tools/util/marshaller.go

```diff
@@ -45,6 +45,30 @@ func MarshallObject(obj interface{}, writer io.Writer) error {
 	unstructured.RemoveNestedField(r.Object, "spec", "template", "metadata", "creationTimestamp")
 	unstructured.RemoveNestedField(r.Object, "status")
 
+	// remove dataSource from PVCs if empty
+	templates, exists, err := unstructured.NestedSlice(r.Object, "spec", "dataVolumeTemplates")
+	if exists {
+		for _, tmpl := range templates {
+			template := tmpl.(map[string]interface{})
+			_, exists, err = unstructured.NestedString(template, "spec", "pvc", "dataSource")
+			if !exists {
+				unstructured.RemoveNestedField(template, "spec", "pvc", "dataSource")
+			}
+		}
+		unstructured.SetNestedSlice(r.Object, templates, "spec", "dataVolumeTemplates")
+	}
+	objects, exists, err := unstructured.NestedSlice(r.Object, "objects")
+	if exists {
+		for _, obj := range objects {
```

Comment:
I get the first loop that checks for dataVolumeTemplates with PVCs and an empty datasource attribute.  I don't understand what's happening in this second loop.

It looks like we're looking for any object with a spec that contains dataSource.  While the chances are small, this could have unintended consequences. Can we ensure this spec belongs to a PVC?

Reply:
Yep, I'll add a check to ensure the spec belongs to a PVC.